src/app/features/genes/components/gene-comparison-tool/gene-comparison-tool.component.ts

```diff
@@ -236,18 +240,41 @@ export class GeneComparisonToolComponent implements OnInit, AVI, OnDestroy {
   getGeneProperty(gene: GCTGene, property: string) {
     return property.split('.').reduce((o: any, i: any) => o[i], gene);
   }
-
+  
   initData(genes: GCTGene[]) {
+    // hide brain region columns initially
     this.brainRegionsColumns.forEach(c => c.visible = false);
-
+    
     const pinnedGenes: GCTGene[] = [];
-    const currentPinnedGenesCache = this.getPinnedGenesCache(
-      this.category,
-      this.subCategory
-    );
-    const urlPins = currentPinnedGenesCache.length
-      ? currentPinnedGenesCache.map((g: GCTGene) => g.uid)
-      : this.getUrlParam('pinned', true);
+    
+    const currentPinnedGenesCache = this.pinnedGenesCache;
+
+    let previousPins: (string | undefined)[] = [];
```

Comment:
Add a comment here noting the previousPins contains the uids of the previously pinned genes, formatted based on the _current_ category. Maybe provide an example? e.g. previous category was Protein, which has uids of ensembl + uniprot, but previousPins should contain just ensembl

Reply:
I will create a helper method for this purpose